### DISCUSSION TITLE
Performance issue

### DISCUSSION THREAD

**Discussion by roganov** [microsoft/pyright#7612] [General]
Hello

I'm adding type annotations to a project and have been using MyPy. I decided to give Pyright a go, but it turned out to be extremely slow:
```
Completed in 1680.611sec
```
MyPy takes several seconds even without a cache.

Unfortunately, I do not have a repro given that this is a large proprietary codebase, otherwise I would've opened an issue. There are a lot of typed dictionaries, which as I understand may be hard to type check, but I can't pinpoint the problem. The verbose mode doesn't seem to report anything useful, just a bunch of `Long operation: analyzing` and `Long operation: checking`.

Is there any way I can provide useful data given that I can't show the code?

pyright 1.1.357

**Comment by erictraut** [comment]
I'd be happy to diagnose the issue, but I would need access to the code to do so. Diagnosing performance issues like this can be difficult and time-consuming even with a repro. Without a repro, it's almost impossible.

One thing that may help you narrow down the issue is to run the command-line version with both the `--stats` and `--verbose` options. With both of these enabled, pyright will dump a table of all type-checked source files sorted by the time it took to check each (from highest to lowest). Typically, you'll find that there's one or two files that dominate that list. From there, you can bisect the file (commenting out portions of it) until you get to a particular function or method.

**Comment by roganov** [comment]
Thank you, I've opened https://github.com/microsoft/pyright/issues/7617. I'm pretty sure there's another issue with an overloaded generic function, but I wasn't able to contain it yet.

**Comment by roganov** [comment]
Resolved with the new release.
